src/zircon/apis/chunkserver.go

```diff
@@ -0,0 +1,76 @@
+package apis
+
+import "time"
+
+// The version number of a chunk
+type Version uint64
+// An offset within a chunk
+type Offset uint32
+// A length of data within a chunk
+type Length uint32
+// A chunk identifier, not directly exposed to normal clients
+type ChunkNum uint64
+
+// 8 MiB, the maximum size of a chunk stored on the chunkserver
+const MaxChunkSize Length = 8 * 1024 * 1024
+
+// Represents "any version is valid" when passed as a chunk version number
+const AnyVersion Version = 0
+
+// An opaque reference to a particular chunk.
+// Does not include which server to look on, but does include the chunk ID.
+// This acts as a bearer token -- whoever has a copy of it has permission to perform operations on this chunk.
+type ChunkReference struct {
+	Chunk       ChunkNum
+	IsReadWrite bool
+	// TODO: figure this security part out
+	Expiration  time.Time
+	Signature   []byte
+}
+
+// note: this API is strongly consistent, because it's a connection to just a single chunkserver
+type Chunkserver interface {
+	// ** methods used by clients and metadata caches **
+
+	// Given a chunk reference, read out part or all of a chunk.
+	// If 'minimum' is AnyVersion, then whichever version the chunkserver currently has will be returned.
+	// If the version of the chunk that this chunkserver has is at least the minimum version, it will be returned.
+	// Otherwise, an error will be returned, along with the most recent available version.
+	// The sum of offset + length must not be greater than MaxChunkSize. The number of bytes returned is always exactly
+	// the same number of bytes requested, unless an error condition is signaled.
+	// The version of the data actually read will be returned.
+	// Fails if a copy of this chunk isn't located on this chunkserver.
+	Read(ref ChunkReference, offset Offset, length Length, minimum Version) ([]byte, Version, error)
+
+	// Given a chunk reference, begin a write to that chunk.
+	// The sum of 'offset' and 'len(data)' must not be greater than MaxChunkSize.
+	// If replicas is nonempty, this will also replicate the prepared write to those servers.
+	// Fails if a copy of this chunk isn't located on this chunkserver, or if another server fails to prepare.
+	StartWrite(ref ChunkReference, offset Offset, data []byte, replicas []ServerAddress) (error)
```

Comment:
It's not just Write, because we need to wait until all three have received the data, not just one.

It includes specific replicas, because the chunkserver isn't smart, and shouldn't have to load the entire metadata block into memory to figure out how to complete a write request.